src/main/java/ch/wisv/areafiftylan/service/TeamServiceImpl.java

```diff
@@ -109,12 +110,17 @@ public TeamInviteToken inviteMember(Long teamId, String username) {
         User user = userService.getUserByUsername(username).orElseThrow(() -> new UsernameNotFoundException(username));
         Team team = getTeamById(teamId);
 
+        // Check if the member isn't already part of the team
+        if (team.getMembers().contains(user)) {
+            throw new IllegalArgumentException("User is already a member of this team");
+        }
+
         // Check if the member isn't already invited
-        Optional<TeamInviteToken> optionalInvite = teamInviteTokenRepository.findByUserUsername(username).stream().
-                filter(token -> token.getTeam().equals(team)).findFirst();
+        boolean userAlreadyInvited = teamInviteTokenRepository.findByUserUsername(username).stream()
```

Comment:
I would extract this logic (as well as some other logic in this method) in seperate methods. This entire method is 30 lines long. In my opinion way too long. This does not only provide better code readability but also code reusability.